x-pack/legacy/plugins/uptime/server/lib/adapters/pings/elasticsearch_pings_adapter.ts

```diff
@@ -163,17 +175,27 @@ export class ElasticsearchPingsAdapter implements UMPingsAdapter {
     request: any,
     dateRangeStart: string,
     dateRangeEnd: string,
-    filters?: string | null
+    filters?: string | null,
+    monitorId?: string | null
   ): Promise<HistogramDataPoint[]> {
     const { statusFilter, query } = getFilteredQueryAndStatusFilter(
       dateRangeStart,
       dateRangeEnd,
       filters
     );
+
+    const combinedQuery = !monitorId
+      ? query
+      : {
+          bool: {
+            must: [{ match: { 'monitor.id': monitorId } }, query],
```

Comment:
I've switched it over to `filter` instead of `must`, but I think this approach is cleaner, and logically identical. I'd rather not reach into the JSON object here, and would rather let ES handle the work, which it does fine.

Reply:
👍